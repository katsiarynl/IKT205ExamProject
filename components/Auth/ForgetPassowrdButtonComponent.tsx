import { View, Text } from "react-native";
import React from "react";
import { TouchableOpacity } from "react-native-gesture-handler";
import { useNavigation } from "@react-navigation/native";
import { NavigationSignOut } from "../../types/navigationTypes";
import { forgetPassStyle } from "../../styles/forgetPasswor";

export default function ForgetPassowrdButtonComponent() {
  const navigation = useNavigation<NavigationSignOut>();

  return (
    <View style={forgetPassStyle.container}>
      <TouchableOpacity
        testID="ForgetPasswordButton"
        onPress={() => navigation.navigate("ForgetPass")}
      >
        <Text style={{ fontWeight: "bold", color: "#7E3B14", fontSize: 19 }}>
          Forget Password?{" "}
        </Text>
      </TouchableOpacity>
    </View>
  );
}
