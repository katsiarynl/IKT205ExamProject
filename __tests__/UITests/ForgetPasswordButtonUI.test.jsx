import { NavigationContainer } from "@react-navigation/native";
import ForgetPassowrdButtonComponent from "../../components/Auth/ForgetPassowrdButtonComponent";
import { create } from "react-test-renderer";

const mockedNavigate = jest.fn();
//https://stackoverflow.com/questions/61781274/how-to-mock-usenavigation-hook-in-react-navigation-5-0-for-jest-test
jest.mock("@react-navigation/native", () => ({
  useNavigation: () => ({ navigate: mockedNavigate }),
}));

const tree = create(<ForgetPassowrdButtonComponent />);

test("snapshot", () => {
  expect(tree).toMatchSnapshot();
});
