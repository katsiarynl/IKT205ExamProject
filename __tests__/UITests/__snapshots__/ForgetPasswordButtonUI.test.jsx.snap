// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginTop": 20,
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    testID="ForgetPasswordButton"
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#7E3B14",
            "fontSize": 19,
            "fontWeight": "bold",
          }
        }
      >
        Forget Password?
         
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;
